lib/dictBuilder/cover.c

```diff
@@ -929,6 +929,108 @@ void COVER_best_finish(COVER_best_t *best, size_t compressedSize,
   }
 }
 
+COVER_dictSelection_t COVER_dictSelectionError(size_t error) {
+    const COVER_dictSelection_t selection = { NULL, 0, error };
+    return selection;
+}
+
+unsigned COVER_dictSelectionIsError(COVER_dictSelection_t selection) {
+  return (ZSTD_isError(selection.totalCompressedSize) || !selection.dictContent);
+}
+
+COVER_dictSelection_t COVER_selectDict(void* customDictContent,
+        size_t dictContentSize, const void* samplesBuffer, const size_t* samplesSizes, size_t nbFinalizeSamples,
+        size_t nbCheckSamples, size_t nbSamples, ZDICT_cover_params_t params, size_t* offsets, size_t totalCompressedSize) {
+
+  size_t largestDict = 0;
+  size_t largestCompressed = 0;
+  double regressionTolerance = 1 + ((double)params.shrinkDict / 100);
+  char* customDictContentEnd = ((char*)customDictContent + dictContentSize);
+
+  BYTE *const dict = (BYTE * const)malloc(dictContentSize);
```

Comment:
nit: Lets have better names than `dict` and `dictBuffer`.

Maybe `largestDictbuffer` and `smallerDictBuffer` or `candidateDictBuffer`?